### PR TITLE
node: skip link checker if no docs are generated

### PR DESCRIPTION
**Pull request by kinwa91**
The dead link checker can now be run with `node doc-check.js` or by creating an npm run-script and execute via `npm run doc-check`.

This is so that we can skip running doc checker if the repository doesn't generate any docs by specifying a no-op `doc-check` script:

```json
  "docs": "no docs yet!",
  "doc-check": "no docs yet"
```

I will propagate using `repo apply` once this looks ok.

### CODE REVIEW

File: synthtool/gcp/templates/node_library/doc-check.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/*!
```
[the remark this answers is not in the document]
Comment (JustinBeckwith) [reply synthtool/gcp/templates/node_library/doc-check.js]: Meep.  I'd rather not do this :/ . Isn't easier to just generate docs?

[the remark this answers is not in the document]
Comment (kinwa91) [reply synthtool/gcp/templates/node_library/doc-check.js]: Oh this just basically do the same thing where we used to do in `docs.sh` to check dead links  🤔, except we can choose to skip this check. Sorry if I'm misunderstanding but what are we concerned about here?

[the remark this answers is not in the document]
Comment (kinwa91) [reply synthtool/gcp/templates/node_library/doc-check.js]: You're right, this doesn't work so well with the additional dependencies, etc..

**Comment by JustinBeckwith** [comment]
I think we can just drop this.  Making it generate docs ain't hard

**Pull request closed without merging**